app/admin/contributions.rb

```diff
@@ -15,12 +15,17 @@
   member_action :refuse, method: :put, only: :index
 
   batch_action :refuse, if: proc { params[:scope] != "recusado" && params[:scope] != "aprovado" } do |ids|
-    batch_action_collection.find(ids).each { |contribution| contribution.refuse!(current_user.id) if contribution.state == "received" }
+    batch_action_collection.find(ids).each { |contribution|
+      if contribution.received?
+        contribution.refuse!(current_user.id)
+        contribution.update(rejected_reason: :other_reason)
```

Comment:
I believe it should be inside of a Transaction to avoid inconsistencies. 

WDYT @adbatista ?

Reply:
Done